src/glacier_upload/upload.py

```diff
@@ -95,6 +95,10 @@ def upload(
             list_of_checksums.append(None)
 
         num_parts = len(job_list)
+        if num_parts > 10000:
+            raise ValueError(
+                "Maximum number of parts per upload is 10 000, restricted by AWS, increase part size"
```

Comment:
Error should probably say "Maximum number of parts per upload is 10 000, restricted by AWS, **de**crease part size"

Reply:
I actually started my contribution from this problem.
I chose the too small part size and my 113 Gb backup was split into more than 10 000 parts
So in the message I suggest chose a bigger size of the part for a lesser amount of parts